osdep/mac/presentation.swift

```diff
@@ -0,0 +1,64 @@
+/*
+ * This file is part of mpv.
+ *
+ * mpv is free software) you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation) either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * mpv is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY) without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+extension Presentation {
+    struct Time {
+        let cvTime: CVTimeStamp
+        var skipped: Int64 = 0
+        var time: Int64 { return mp_time_ns_convert(Int64(Double(cvTime.hostTime) * Presentation.machToNano)) }
```

Comment:
To avoid duplication of code, I would extend `timer-darwin.c` with `mp_time_from_mach` and use it here, instead `mp_time_ns_convert` and `Presentation.machToNano` which is already calculated in `timer-darwin.c`

Reply:
yeah we have some duplications here and in precise_timer.swift. i probably need to clean this up a bit more in the future.

i will look into your suggestions here and your comment on osdep/timer.c.